File: backend/fief/schemas/user_role.py
from pydantic import UUID4

from fief.schemas.generics import BaseModel, CreatedUpdatedAt
from fief.schemas.role import RoleEmbedded


class UserRoleCreate(BaseModel):
    id: UUID4


class BaseUserRole(CreatedUpdatedAt):
    user_id: UUID4
    role_id: UUID4


class UserRole(BaseUserRole):
    role: RoleEmbedded

File: admin-frontend/src/schemas/user-role.ts
import { CreatedUpdatedAt } from './generics';
import { RoleEmbedded } from './role';

export interface UserRoleCreate {
  id: string;
}

interface BaseUserRole extends CreatedUpdatedAt {
  user_id: string;
  role_id: string;
}

export interface UserRole extends BaseUserRole {
  role: RoleEmbedded;
}
